apps/ex_nvr_web/lib/ex_nvr_web/live/onvif_discovery_live.ex

```diff
@@ -82,15 +82,47 @@ defmodule ExNVRWeb.OnvifDiscoveryLive do
           |> fetch_network_settings(device.url, opts)
           |> fetch_media_profiles(media_url, opts)
           |> fetch_stream_uris(media_url, opts)
+          |> fetch_snapshot_uris(media_url, opts)
 
         {:noreply,
          assign(socket,
+           selected_device: device,
            device_details: device_details,
            device_details_cache: Map.put(device_details_cache, device_name, device_details)
          )}
     end
   end
 
+  def handle_event("add-device", _params, socket) do
+    selected_device = socket.assigns.selected_device
+    device_details = socket.assigns.device_details
+
+    %{stream_uri: stream_uri} = Enum.at(device_details.media_profiles, 0, %{stream_uri: nil})
+    %{stream_uri: sub_stream_uri} = Enum.at(device_details.media_profiles, 1, %{stream_uri: nil})
+
+    %{snapshot_uri: snapshot_uri} =
+      Enum.at(device_details.media_profiles, 0, %{snapshot_uri: nil})
+
+    socket
+    |> put_flash(:device_params, %{
+      name: selected_device.name,
+      type: :ip,
+      vendor: get_vendor_code(device_details.infos.manufacturer),
+      stream_config: %{
+        stream_uri: stream_uri,
+        sub_stream_uri: sub_stream_uri,
+        snapshot_uri: snapshot_uri
+      },
+      credentials: %{username: selected_device.username, password: selected_device.password}
+    })
+    |> redirect(to: ~p"/devices/new")
+    |> then(&{:noreply, &1})
+  end
+
+  defp get_vendor_code("Milesight Technology Co.,Ltd."), do: "Milesight Technology Co.,Ltd"
+  defp get_vendor_code("HIKVISION"), do: "Hikvision"
```

Comment:
Whatever you get from the onvif response is the right answer. Update old list to match this values.